### PR TITLE
Auto-format all files with yapf Google style.

### LINKED ISSUE #21: Auto-format Python code
Per discussion in #20, auto-format all Python code using https://github.com/google/yapf and "google" style.  Set up a Makefile target for easy formatting (and CI testing).

### PR DESCRIPTION
**Pull request by nicksay**
See https://github.com/google/yapf for more information.

Closes #21

### CODE REVIEW

**Comment by awbraunstein** [comment]
I'm going to assume all the tests still pass. lgtm.

**Comment by nicksay** [comment]
The unit tests pass, but I'm going to wait until the full test suit is fixed, just in case (see #26).

**Comment by nicksay** [comment]
@awbraunstein This is ready for another look

File: spitfire/compiler/analyzer.py [hunk 1]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+        self.parse_root = parse_root
+        self.options = options
+        self.compiler = compiler
+        self.ast_root = None
+        self.template = None
+        self.strip_lines = False
+        self.uses_raw = False
+        self.base_extends_identifiers = []
+        if self.compiler.base_extends_package:
+            # this means that extends are supposed to all happen relative to some
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: These comments are now longer than 80 cols. Can you rejustify?

File: spitfire/compiler/analyzer.py [hunk 2]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+        # that extends unicode or str. Depending on the mode, it would
+        # assign SanitizedPlaceholder to be the correct class.
+        if self.options.generate_unicode and self.options.baked_mode:
+            self.compiler.error(
+                SemanticAnalyzerError(
+                    'Generate unicode is incompatible with baked mode.'),
+                pos=pnode.pos)
+        self.template.baked = self.options.baked_mode
+
+        # Need to build a full list of template_methods before analyzing so we can
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: longer than 80 cols

File: spitfire/compiler/analyzer.py [hunk 3]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+        self.template.main_function.child_nodes = self.optimize_buffer_writes(
+            self.template.main_function.child_nodes)
+
+        if self.template.extends_nodes and self.template.library:
+            self.compiler.error(SemanticAnalyzerError(
+                "library template can't have extends."))
+
+        return [self.template]
+
+    # Recursively grabs identifiers from a ast.TargetListNode, such as in a ast.ForNode.
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: longer than 80 cols

File: spitfire/compiler/analyzer.py [hunk 4]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+            elif ph_expression.library_function:
+                # Don't escape function calls into library templates.
+                skip_filter = True
+
+        if (self.compiler.enable_filters and
+                format_string == default_format_string and
+                not isinstance(ph_expression, ast.LiteralNode)):
+            arg_node_map = pnode.parameter_list.get_arg_node_map()
+            if 'raw' in arg_map:
+                # If this is a |raw usage and the template does not allow raw, raise an
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: 80 cols

File: spitfire/compiler/analyzer.py [hunk 5]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+                # error.
+                self.uses_raw = True
+                if self.options.no_raw and not self.template.explicitly_allow_raw:
+                    self.compiler.error(
+                        SemanticAnalyzerError(
+                            '|raw is not allowed in templates compiled with the '
+                            '--no-raw flag.'),
+                        pos=pnode.pos)
+            else:
+                # if we need to filter, wrap up the node and wait for further analysis
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: 80 cols

File: spitfire/compiler/analyzer.py [hunk 6]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+                # later on
+                if skip_filter:
+                    # explicitly set the filter to none here - this means we will cache
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: 80 cols

File: spitfire/compiler/analyzer.py [hunk 7]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+                                                   None,
+                                                   pos=pnode.pos)
+                else:
+                    ph_expression = ast.FilterNode(
+                        ph_expression,
+                        arg_node_map.get('filter', ast.DefaultFilterFunction),
+                        pos=pnode.pos)
+
+                # if this is a literal node, we still might want to filter it
+                # but the output should always be the same - so do it once and cache
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: 80 cols

File: spitfire/compiler/analyzer.py [hunk 8]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+        return node_list
+
+    def analyzePlaceholderNode(self, pnode):
+        if (self.options.fail_library_searchlist_access and
+                pnode.name not in self.template.global_placeholders):
+            if (self.options.strict_global_check and
+                    not self.template.allow_undeclared_globals and
+                (not self.template.has_identifier(pnode.name) and
+                 pnode.name not in self.compiler.function_name_registry)):
+                # Break compile if no #loose_resolution and variable is not available
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: 80 cols

File: spitfire/compiler/analyzer.py [hunk 9]
```diff
@@ -64,698 +58,720 @@ class MacroParseError(MacroError):
...
+                # in any reasonable scope.
+                self.compiler.error(
+                    SemanticAnalyzerError(
+                        'identifier %s is unavailable and is not declared as a #global'
+                        ' display variable' % pnode.name),
+                    pos=pnode.pos)
+            elif self.template.library:
+                # Only do placeholder resolutions for placeholders declared with #global
```
Comment (awbraunstein) [inline spitfire/compiler/analyzer.py]: 80 cols

File: spitfire/compiler/analyzer.py (hunk 1 above)
Comment (nicksay) [inline spitfire/compiler/analyzer.py]: Done

File: spitfire/compiler/analyzer.py (hunk 2 above)
Comment (nicksay) [inline spitfire/compiler/analyzer.py]: Done

File: spitfire/compiler/analyzer.py (hunk 3 above)
Comment (nicksay) [inline spitfire/compiler/analyzer.py]: Done

File: spitfire/compiler/analyzer.py (hunk 4 above)
Comment (nicksay) [inline spitfire/compiler/analyzer.py]: Done

File: spitfire/compiler/analyzer.py (hunk 1 above)
Comment (nicksay) [inline spitfire/compiler/analyzer.py]: Done

**Comment by nicksay** [comment]
I've adjusted all comments and docstrings to 80 chars.  I've left the update as a second commit but will rebase before merging.

**Comment by nicksay** [comment]
I've gone ahead and squashed the commits.

**Comment by awbraunstein** [comment]
lgtm!! :panda_face:

**Pull request merged**